### ISSUE TITLE
Splash Screen Access Violation in Skia4Delphi

### ISSUE THREAD

**Issue by TheOriginalBytePlayer** [skia4delphi/skia4delphi#188]
I upgraded to the latest version of Skia4Delphi, and when I launch my app within the debugger, my splash screen almost immediately causes an access violation in Skia.dll

![image](https://user-images.githubusercontent.com/18058224/206824403-01c1656c-8fbc-42cf-a0d9-9917ac1d9180.png)

If I run it outside of the debugger I don't get a reported access violation but it looks completely scrambled:

![image](https://user-images.githubusercontent.com/18058224/206824523-44d83d7b-b7ba-43db-8b5e-e372db19963c.png)

versus what it should look like (and what it looked like prior to upgrading), which is 

![image](https://user-images.githubusercontent.com/18058224/206824541-2112409f-cf6e-4299-b4f6-980b3c174a38.png)

Curiously, outside of the debugger, the actual first form shown created after the mainform has been created (and all subsequent forms) display correctly.

My project uses is:
```delphi
uses
  System.StartUpCopy,
  FMX.Forms,
  Skia.FMX,
  GlobalSkiaInitializer in 'shared\GlobalSkiaInitializer.pas',
  FMX.Types,
  FMX.Graphics,
  FMX.Platform,
etc.
```
and GlobalSkinInitializer.pas contains 
```delphi
unit GlobalSkiaInitializer;

interface

implementation

uses Skia.FMX.Graphics, Skia.FMX;

initialization
  GlobalUseSkia := True;
  GlobalUseSkiaRasterWhenAvailable := True;
end.
```
As I want the Splash Screen to show up as quickly as possible, and DO NOT want it to become my mainform, I am creating it directly using

```delphi
procedure ShowSplashScreen;
begin
  SplashScreen:=TSplashScreen.Create(Nil);
  Show;
  Invalidate;
end;
```
Though the access violation doesn't happen until I call application.processmessages to get it to actually display.  This all worked fine with version 3.14 of Skia4Delphi

Please help as otherwise I"ll have to downgrade my skia4Delphi

**Comment by Spelt** [comment]
My 2 cents: I know application.proccesmessages is something to avoid in Android.  Op 10 dec. 2022 om 03:42 heeft TheOriginalBytePlayer ***@***.***> het volgende geschreven:﻿
I upgraded to the latest version of Skia4Delphi, and when I launch my app within the debugger, my splash screen almost immediately causes an access violation in Skia.dll

If I run it outside of the debugger I don't get a reported access violation but it looks completely scrambled:

versus what it should look like (and what it looked like prior to upgrading), which is

My project uses is:
uses
  System.StartUpCopy,
  FMX.Forms,
  Skia.FMX,
  GlobalSkiaInitializer in 'shared\GlobalSkiaInitializer.pas',
  FMX.Types,
  FMX.Graphics,
  FMX.Platform,
etc.

and GlobalSkinInitializer.pas contains
unit GlobalSkiaInitializer;

interface

implementation

uses Skia.FMX.Graphics, Skia.FMX;

initialization
  GlobalUseSkia := True;
  GlobalUseSkiaRasterWhenAvailable := True;
end.

As I want the Splash Screen to show up as quickly as possible, I am creating it directly using
procedure ShowSplashScreen;
begin
  SplashScreen:=TSplashScreen.Create(Nil);
  Show;
  Invalidate;
end;

Though the access violation doesn't happen until I call application.processmessages to get it to actually display.  This all worked fine with version 3.14 of Skia4Delphi

**Comment by viniciusfbb** [comment]
Hi @TheOriginalBytePlayer! Thank you very much for reporting the problem.

Can you give us more information? Is this AV occurring on Windows? Could you run debug mode and send a print of the callstack when you get AV?

When exactly do you call Application.ProcessMessages? I know it's laborious, but could you provide a simple project that simulates the problem?

**Comment by TheOriginalBytePlayer** [comment]
With the latest build of Delphi 11.3 and the latest build of Skia4Delphi from GetIt, the AV is no longer occurring BUT the scrambled Splash Screen Image is.
Here is a small example that shows it: [Splash Screen Bug Sample.zip](https://github.com/skia4delphi/skia4delphi/files/10865544/Splash.Screen.Bug.Sample.zip)

**Comment by viniciusfbb** [comment]
Fixed in our new release [v5.0.0](https://github.com/skia4delphi/skia4delphi/releases)
